src/main/java/uk/gov/justice/hmpps/prison/service/OffenderTransactionHistoryService.java

```diff
@@ -0,0 +1,61 @@
+package uk.gov.justice.hmpps.prison.service;
+
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.validation.annotation.Validated;
+import uk.gov.justice.hmpps.prison.api.model.OffenderTransactionHistoryDto;
+import uk.gov.justice.hmpps.prison.repository.OffenderTransactionHistoryRepository;
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderTransactionHistory;
+import uk.gov.justice.hmpps.prison.service.transformers.OffenderTransactionHistoryTransformer;
+import uk.gov.justice.hmpps.prison.values.AccountCode;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+@Service
+@Transactional(readOnly = true)
+@AllArgsConstructor
+@Validated
+@Slf4j
+public class OffenderTransactionHistoryService {
+
+    private OffenderTransactionHistoryRepository repository;
+
+    public List<OffenderTransactionHistoryDto> getTransactionHistory(final Long offenderId,
```

Comment:
in my view, a service class should not return a dto, it should return a domain class and it's up to the web controller to convert it to a dto that can be packaged in http response object.

Reply:
There are definitely merits to your argument but looking through prison-api it seems that a pragmatic approach has been adopted to treat the DB layer as the domain layer and we definitely don't want that leaking into the web layer... Plus it would take a lot of work to introduce a domain model between the web and DB layers now.  If we were designing from scratch - it would be an interesting conversation :)